### PR TITLE
Add Klobuchar ionoshperic delay model

### PR DESCRIPTION
**Pull request by ljbade**
Implements the ionospheric delay model from GPS ICD.

Next job will be to extract the parameters from the NAV data and apply the corrections for SPP.

Ready for review.

### CODE REVIEW

File: src/ionosphere.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  }
+
+  /* Compute the longitude of the IPP */
+  double lon_i = lon_u + (psi * sin(a)) / cos(lat_i * M_PI);
+
+  /* Find the geomagnetic latitude of the IPP */
+  double lat_m = lat_i + 0.064 * cos((lon_i - 1.617) * M_PI);
+
+  /* Find the local time at the IPP */
+  double t = 43200.0 * lon_i + t_gps.tow;
```
Comment (fnoble) [inline src/ionosphere.c]: I think you could make better use of the functions in `gpstime.c` here. How about something like
```
gps_time_t t = t_gps;
t.tow += 43200.0 * lon_i;
normalize_gps_time(t);
```

File: tests/check_ionosphere.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#include <check.h>
+
+#include <constants.h>
+#include <ionosphere.h>
+
+START_TEST(test_calc_ionosphere)
+{
+  gps_time_t t = {.wn = 1875, .tow = 479820};
+  ionosphere_t i = {.a0 = 0.1583e-7, .a1 = -0.7451e-8, .a2 = -0.5960e-7, .a3 = 0.1192e-6,
+                    .b0 = 0.1290e6, .b1 = -0.2130e6, .b2 = 0.6554e5, .b3 = 0.3277e6};
```
Comment (fnoble) [inline tests/check_ionosphere.c]: >>80 columns
(yes, I am a pedant)

**Comment by fnoble** [comment]
looks good to me, feel free to merge

any idea what caused the travis test to fail the first time?

**Comment by fnoble** [comment]
should figure that out before merging

File: src/ionosphere.c (hunk 1 above)
Comment (ljbade) [inline src/ionosphere.c]: As per chat, `normalize_gps_time` normalises to a week. This model needs to be normalised to a day.

File: tests/check_ionosphere.c (hunk 2 above)
Comment (ljbade) [inline tests/check_ionosphere.c]: Fixed.

**Comment by ljbade** [comment]
Review comments fixed. Will merge after test pass.

File: include/libswiftnav/constants.h [hunk 3]
```diff
@@ -34,7 +34,7 @@
 /** The official GPS value of Pi.
  * This is the value used by the CS to curve fit ephemeris parameters and
  * should be used in all ephemeris calculations. */
-#define GPS_PI 3.14159265358979323846
+#define GPS_PI 3.1415926535898
```
Comment (akleeman) [inline include/libswiftnav/constants.h]: I'm guessing there was a reason ... but why lower the precision of pi?

Comment (mookerji) [inline include/libswiftnav/constants.h]: I believe the GPS ICD document specifies a lower precision version. See: www.gps.gov/technical/icwg/.

Comment (ljbade) [inline include/libswiftnav/constants.h]: Yep in the IS-GPS-200H, form section 20.3.3.4.3.2:
image

**Comment by ljbade** [comment]
In the future the Galileo correction model should be used for L1.

**Pull request merged**